### PR TITLE
NFC: First step (refactoring) towards speeding up compilation w/ >1 primary file

### PR DESCRIPTION
**Pull request by davidungar**
NFC: First step (refactoring) towards speeding up compilation w/ >1 primary file

Creates new class, FrontendInputs, to encapsulate InputFilenames, InputBuffers, and PrimaryInput, which were formerly in FrontendOptions. Add new instance variable, Inputs, to FrontendOptions in order to hold FrontendInputs.

Encapsulate uses of the variables in FrontendInputs with intention-describing functions. Move some code that sets these variables into FrontendInputs and FrontendOptions classes.

Create new FrontendInputs class to encapsulate InputFilenames, InputBuffers and PrimaryInput, which were formerly in Frontend.

Includes one change in SwiftEditor.cpp to resolve a merge conflict.

### CODE REVIEW

**Comment by davidungar** [comment]
@swift-ci please smoke test

**Comment by davidungar** [comment]
@swift-ci please smoke test

File: include/swift/Frontend/FrontendOptions.h [hunk 1]
```diff
@@ -62,9 +66,11 @@ enum class InputFileKind {
   IFK_LLVM_IR
 };
 
-/// Options for controlling the behavior of the frontend.
-class FrontendOptions {
-public:
+/// Information about all the inputs to the frontend.
+class FrontendInputs {
+private:
+  // FIXME: (dmu) create a class for the inputs
```
Comment (DougGregor) [inline include/swift/Frontend/FrontendOptions.h]: Stale comment?

**Review by DougGregor — APPROVED** [review]
Nice cleanup!

File: include/swift/Frontend/FrontendOptions.h (hunk 1 above)
Comment (davidungar) [reply include/swift/Frontend/FrontendOptions.h]: Thanks. Removed the comment, but forgot to push before a merge.

**Pull request merged**